### PR TITLE
feat: workflows assign request on browser

### PR DESCRIPTION
**Pull request by ayushpahwa**
## Description

We are allowing users to have write workflows assign request queries in the JS object editor for the workflows editor. This means, users should get autocomplete for `appsmith.workflows.assignRequest` and the arguments required should be shown in the autocomplete also. To enable this, following changes have been made
1) Add empty workflows object to the appsmith function type in store and and add ee only type definitions to the `EntityDefinitions` for appsmith namespace.
2) Replace entityFunctions as static variable and use a function which appends EE entity functions.
3) Added types for ee only functions
4) Retain the `isMainJsObject` flag for the js file when updates happen.

#### PR fixes following issue(s)
Fixes # (issue number)

#### Type of change

- New feature (non-breaking change which adds functionality)

- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
	- Introduced new autocomplete definitions for enhanced code editor suggestions.
	- Added support for categorizing entities in the Explorer panel using `GROUP_TYPES` for improved organization and navigation.
- **Refactor**
	- Updated the action helpers and reducers to support new properties and ensure state consistency.
	- Improved the structure for autocomplete helper functions, facilitating easier extension and maintenance.
- **Chores**
	- Established foundational code for future development with empty enums and placeholder functions.

### CODE REVIEW

File: app/client/src/ce/utils/autocomplete/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+import type { Def } from "tern";
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export const eeAppsmithAutocompleteDefs = (generatedTypeDef: Def) => ({});
```
[the remark this answers is not in the document]
Comment (ayushpahwa) [reply app/client/src/ce/utils/autocomplete/helpers.ts:5]: The function definition has to be kept same due to ee only code which uses this argument.

File: app/client/src/ce/utils/autocomplete/EntityDefinitions.ts [hunk 2]
```diff
@@ -88,6 +89,7 @@
               "https://docs.appsmith.com/reference/appsmith-framework/context-object#geolocationclearwatch",
           },
         },
+        ...eeAppsmithAutocompleteDefs(generatedTypeDef),
```
[the remark this answers is not in the document]
Comment (ayushpahwa) [reply app/client/src/ce/utils/autocomplete/EntityDefinitions.ts:92]: The function name indicates that this inputs ee only function definitions to the appsmith function. The actual code for this is available in the EE repo.

File: app/client/src/ce/utils/autocomplete/EntityDefinitions.ts [hunk 3]
```diff
@@ -15,6 +15,7 @@ import type {
 } from "@appsmith/entities/DataTree/types";
 import type { FieldEntityInformation } from "components/editorComponents/CodeEditor/EditorConfig";
 import { AutocompleteDataType } from "utils/autocomplete/AutocompleteDataType";
+import { eeAppsmithAutocompleteDefs } from "@appsmith/utils/autocomplete/helpers";
```
[the remark this answers is not in the document]
Comment (ayushpahwa) [reply app/client/src/ce/utils/autocomplete/EntityDefinitions.ts:18]: The function name indicates that this inputs ee only function definitions to the appsmith function. The actual code for this is available in the EE repo.

File: app/client/src/components/editorComponents/ActionCreator/viewComponents/ActionBlockTree/utils.tsx [hunk 4]
```diff
@@ -88,6 +88,9 @@ function GetIconForAction(
 
     case AppsmithFunction.postWindowMessage:
       return () => <Icon name="chat-upload-icon" />;
+
+    default:
```
Comment (ApekshaBhosale) [inline app/client/src/components/editorComponents/ActionCreator/viewComponents/ActionBlockTree/utils.tsx:92]: is this intended? @ayushpahwa

**Review by ApekshaBhosale — CHANGES_REQUESTED** [review]

File: app/client/src/components/editorComponents/ActionCreator/viewComponents/ActionBlockTree/utils.tsx (hunk 4 above)
Comment (ayushpahwa) [reply app/client/src/components/editorComponents/ActionCreator/viewComponents/ActionBlockTree/utils.tsx:92]: Yup, added a default case for backup.

**Review by ApekshaBhosale — DISMISSED** [review]

**Review by rishabhrathod01 — APPROVED** [review]
LGTM

**Pull request merged**